src/yz_app.erl

```diff
@@ -58,16 +60,46 @@ start(_StartType, _StartArgs) ->
             maybe_setup(Enabled),
 
             %% Now everything is started, permit usage by KV/query
-            yokozuna:enable(index),
-            yokozuna:enable(search),
+            enable_components(),
             {ok, Pid};
         Error ->
             Error
     end.
 
+%% @doc Prepare to stop - called before the supervisor tree is shutdown
+prep_stop(State) ->
+    try %% wrap with a try/catch - application carries on regardless,
+        %% no error message or logging about the failure otherwise.
+        lager:info("Stopping application yokozuna.\n", []),
+        ok = riak_api_pb_service:deregister(?QUERY_SERVICES),
+        ok = riak_api_pb_service:deregister(?ADMIN_SERVICES),
+        ok = yz_solrq_drain_mgr:drain(),
+        ok = disable_components()
+    catch
+        Type:Reason ->
+            lager:error("Stopping application yokozuna - ~p:~p.\n",
+                        [Type, Reason])
+    end,
+    State.
+
 stop(_State) ->
-    ok = riak_api_pb_service:deregister(?QUERY_SERVICES),
-    ok = riak_api_pb_service:deregister(?ADMIN_SERVICES),
+    lager:info("Stopped application yokozuna.\n", []),
```

Comment:
These lager lines have \newlines appended to them, but I don't think they are necessary (from what little I know about lager.  We should check with @macintux or @mrallen1.  Also, I'd prefer we use ~n, as \n is not platform independent.  (Yes, we don't support Windows today, but who knows?)